arlas-core/src/main/java/io/arlas/server/services/ExploreServices.java

```
@@ -394,71 +406,101 @@ public AggregationResponse formatAggregationResult(MultiBucketsAggregation aggre
+                                            if (geometry != null) {
+                                                returnedGeometry = new ReturnedGeometry();
+                                                geometryType = collection.params.getGeometryType(g);
+                                                returnedGeometry.reference = g;
+                                            } else {
+                                                geometry = MapExplorer.getObjectFromPath(collection.params.centroidPath, source);
+                                                geometryType = collection.params.getGeometryType(collection.params.centroidPath);
+                                                if (geometry != null) {
+                                                    returnedGeometry = new ReturnedGeometry();
+                                                    returnedGeometry.reference = collection.params.centroidPath;
+                                                }
+                                            }
+                                            geometryGeoJson = geometry != null ?
+                                                    GeoTypeMapper.getGeoJsonObject(geometry, geometryType) : null;
+                                            if (geometryGeoJson != null) {
+                                                returnedGeometry.geometry = geometryGeoJson;
+                                                returnedGeometry.isRaw = true;
+                                                if (element.geometries == null) {
+                                                    element.geometries = new ArrayList<>();
+                                                }
+                                                element.geometries.add(returnedGeometry);
+                                            }
+                                        } catch (ArlasException e) {
+                                            e.printStackTrace();
```

Won't the exception be hidden?

---

Yes totally. This is a hack to keep rendering the geojson even when we can't parse an unkown format of the indexed geometry.

It should be treated in another PR 

I'll create the issue